### PR TITLE
Add rule S4136 (based on '@typescript-eslint/adjacent-overload-signatures') for TS

### LINKED ISSUE #1462: Add rule S4136 (based on '@typescript-eslint/adjacent-overload-signatures') for TS

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**
Fixes #1462

### Changes in the rule
* Reporting is now done on the last overloaded method.

### CODE REVIEW

File: its/ruling/src/test/expected/ts/angular/typescript-S4136.json [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
+56,
+],
+'angular:packages/platform-webworker/src/web_workers/worker/platform_location.ts':[
+82,
+],
 'angular:packages/upgrade/src/common/angular1.ts':[
-48,
+53,
 ],
-'angular:packages/upgrade/static/src/common/angular1.ts':[
```
Comment (quentin-jaquier-sonarsource) [inline its/ruling/src/test/expected/ts/angular/typescript-S4136.json]: This was a duplicate due to symbolic link. (removed [here](https://github.com/SonarSource/typescript-test-sources/commit/7fde63c53049cd6bd8b356bf8375c43c093bdad5)).

File: its/ruling/src/test/expected/ts/angular/typescript-S4136.json [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
+],
+'angular:packages/platform-browser/src/browser/location/browser_platform_location.ts':[
+56,
+],
+'angular:packages/platform-webworker/src/web_workers/worker/platform_location.ts':[
+82,
+],
 'angular:packages/upgrade/src/common/angular1.ts':[
-48,
+53,
```
Comment (quentin-jaquier-sonarsource) [inline its/ruling/src/test/expected/ts/angular/typescript-S4136.json]: In addition, the line reported seems to have changed.

**Comment by yassin-kammoun-sonarsource** [comment]
Indeed. I don't why I call those FPs.

File: its/ruling/src/test/expected/ts/angular/typescript-S4136.json (hunk 2 above)
Comment (yassin-kammoun-sonarsource) [reply its/ruling/src/test/expected/ts/angular/typescript-S4136.json]: I will update the above comment accordingly.

**Pull request merged**